examples/fluids/qfunctions/shocktube.h

```diff
@@ -0,0 +1,510 @@
+
+    // The Physics
+    // Zero v and dv so all future terms can safely sum into it
+    for (int j=0; j<5; j++) {
+      v[j][i] = 0;
+      for (int k=0; k<3; k++)
+        dv[k][j][i] = 0;
+    }
+
+    // -- Density
+    // ---- u rho
+    for (int j=0; j<3; j++)
+      dv[j][0][i]  += wdetJ*(rho*u[0]*dXdx[j][0] + rho*u[1]*dXdx[j][1] +
+                             rho*u[2]*dXdx[j][2]);
+    // -- Momentum
+    // ---- rho (u x u) + P I3
+    for (int j=0; j<3; j++)
+      for (int k=0; k<3; k++)
+        dv[k][j+1][i]  += wdetJ*((rho*u[j]*u[0] + (j==0?P:0))*dXdx[k][0] +
+                                 (rho*u[j]*u[1] + (j==1?P:0))*dXdx[k][1] +
+                                 (rho*u[j]*u[2] + (j==2?P:0))*dXdx[k][2]);
+    // -- Total Energy Density
+    // ---- (E + P) u
+    for (int j=0; j<3; j++)
+      dv[j][4][i]  += wdetJ * (E + P) * (u[0]*dXdx[j][0] + u[1]*dXdx[j][1] +
+                                         u[2]*dXdx[j][2]);
+
+    // -- YZB stabilization
+    if (context->yzb) {
+      CeedScalar drho_norm = 0.0;   // magnitude of the density gradient
+      CeedScalar j_vec[3] = {0.0};  // unit vector aligned with the density gradient
+      CeedScalar jdgn[3] = {0.0};   // j * grad(N)
+      CeedScalar h_shoc = 0.0;      // element lengthscale
+      CeedScalar u_cha = 0.0;       // characteristic velocity, set to acoustic speed
+      CeedScalar tau_shoc = 0.0;    // timescale
+      CeedScalar nu_shoc = 0.0;     // artificial diffusion
+
+      // Unit vector aligned with the density gradient
+      drho_norm = sqrt(drhodx[0]*drhodx[0] + drhodx[1]*drhodx[1] +
+                       drhodx[2]*drhodx[2]);
+      for (int j=0; j<3; j++)
+        j_vec[j] = drhodx[j] / (drho_norm + 1e-20);
+
+      // Approximate dot(j_vec,grad(N)) using the metric tensor
+      for (int j=0; j<3; j++)
+        jdgn[j] = j_vec[0] * dXdx[0][j]
+                  + j_vec[1] * dXdx[1][j]
+                  + j_vec[2] * dXdx[2][j];
+
+      if (drho_norm == 0.0) {
+        nu_shoc = 0.0;
+      } else {
+        h_shoc = 2.0 / (Cyzb * sqrt(jdgn[0]*jdgn[0] + jdgn[1]*jdgn[1] +
+                                    jdgn[2]*jdgn[2]));
+        u_cha = sqrt(gamma*P/rho);
+        tau_shoc = h_shoc / (2*u_cha) * pow(drho_norm * h_shoc / rho, Byzb);
+        nu_shoc = fabs(tau_shoc * u_cha * u_cha);
+      }
+
+      for (int j=0; j<3; j++)
+        dv[j][0][i] -= wdetJ * nu_shoc * drhodx[j];
+
+      for (int k=0; k<3; k++)
+        for (int j=0; j<3; j++)
+          dv[j][k][i] -= wdetJ * nu_shoc * du[k][j];
+
+      for (int j=0; j<3; j++)
+        dv[j][4][i] -= wdetJ * nu_shoc * dEdx[j];
+    }
+
+    // Stabilization
+
+    // Need the Jacobian for the advective fluxes for stabilization
+    //    indexed as: jacob_F_conv[direction][flux component][solution component]
+    CeedScalar jacob_F_conv[3][5][5] = {{{0.}}};
+    CeedScalar velsq = u[0]*u[0] + u[1]*u[1] + u[2]*u[2];
+    CeedScalar rhosq = rho*rho;
```

Comment:
libCEED's convention is to use full words in general, which underscores between them, so something like `vel_sq` or `velocity_sq`